pkg/apis/v1/nodepool_conversion.go

```diff
@@ -0,0 +1,167 @@
+			return nc.Kind == in.Spec.NodeClassRef.Kind && nc.Group == in.Spec.NodeClassRef.Group
+		})
+		v1beta1np.Spec.NodeClassRef = &v1beta1.NodeClassReference{
+			Kind:       in.Spec.NodeClassRef.Kind,
+			Name:       in.Spec.NodeClassRef.Name,
+			APIVersion: lo.Ternary(found, nodeclass.GroupVersion().String(), ""),
+		}
+	}
+
+	// Need to implement Kubelet Conversion
+}
+
+// Convert v1beta1 NodePool to V1 NodePool
+func (in *NodePool) ConvertFrom(ctx context.Context, v1beta1np apis.Convertible) error {
+	v1beta1NP := v1beta1np.(*v1beta1.NodePool)
+	in.Name = v1beta1NP.Name
+	in.UID = v1beta1NP.UID
+	in.Annotations = v1beta1NP.Annotations
+	in.Labels = v1beta1NP.Labels
+
+	// Convert v1beta1 status
+	in.Status.Resources = v1beta1NP.Status.Resources
+
+	in.Spec.convertFrom(ctx, &v1beta1NP.Spec)
+	in.Annotations = lo.Assign(in.Annotations, map[string]string{
+		NodePoolHashVersion:              in.Hash(),
+		NodePoolHashVersionAnnotationKey: NodePoolHashVersion,
+	})
+	return nil
+}
+
+func (in *NodePoolSpec) convertFrom(ctx context.Context, v1beta1np *v1beta1.NodePoolSpec) {
+	in.Weight = v1beta1np.Weight
+	in.Limits = Limits(v1beta1np.Limits)
+	in.Disruption.convertFrom(&v1beta1np.Disruption)
+	in.Template.convertFrom(ctx, &v1beta1np.Template)
+}
+
+func (in *Disruption) convertFrom(v1beta1np *v1beta1.Disruption) {
+	in.ConsolidateAfter = (*NillableDuration)(v1beta1np.ConsolidateAfter)
+	in.ConsolidationPolicy = ConsolidationPolicy(v1beta1np.ConsolidationPolicy)
+	in.ExpireAfter = NillableDuration(v1beta1np.ExpireAfter)
+	in.Budgets = lo.Map(v1beta1np.Budgets, func(v1beta1Budget v1beta1.Budget, _ int) Budget {
+		return Budget{
+			Nodes:    v1beta1Budget.Nodes,
+			Schedule: v1beta1Budget.Schedule,
+			Duration: v1beta1Budget.Duration,
+		}
+	})
+}
+
+func (in *NodeClaimTemplate) convertFrom(ctx context.Context, v1beta1np *v1beta1.NodeClaimTemplate) {
+	in.Annotations = v1beta1np.Annotations
+	in.Labels = v1beta1np.Labels
+	in.Spec.Taints = v1beta1np.Spec.Taints
+	in.Spec.StartupTaints = v1beta1np.Spec.StartupTaints
+	in.Spec.Requirements = lo.Map(v1beta1np.Spec.Requirements, func(v1beta1Requirements v1beta1.NodeSelectorRequirementWithMinValues, _ int) NodeSelectorRequirementWithMinValues {
+		return NodeSelectorRequirementWithMinValues{
+			NodeSelectorRequirement: v1.NodeSelectorRequirement{
+				Key:      v1beta1Requirements.Key,
+				Values:   v1beta1Requirements.Values,
+				Operator: v1beta1Requirements.Operator,
+			},
+			MinValues: v1beta1Requirements.MinValues,
+		}
+	})
+
+	nodeclasses := injection.GetNodeClasses(ctx)
+	if v1beta1np.Spec.NodeClassRef != nil {
+		in.Spec.NodeClassRef = &NodeClassReference{
+			Name:  v1beta1np.Spec.NodeClassRef.Name,
+			Kind:  lo.Ternary(v1beta1np.Spec.NodeClassRef.Kind == "", nodeclasses[0].Kind, v1beta1np.Spec.NodeClassRef.Kind),
+			Group: lo.Ternary(v1beta1np.Spec.NodeClassRef.APIVersion == "", nodeclasses[0].Group, strings.Split(v1beta1np.Spec.NodeClassRef.APIVersion, "/")[0]),
```

Comment:
I think there's an idiomatic function to parse the APIVersion in `apimachinery`. I think it basically boils down to the same thing that you have here, but just a callout if you want to use that instead